lib/cisco_node_utils/cisco_cmn_utils.rb

```diff
@@ -224,5 +224,127 @@ def self.array_to_str(array, sort=true)
       end
       lranges.to_s.gsub('..', '-').delete('[').delete(']').delete(' ')
     end
+
+    # normalize_range_array
+    #
+    # Given a list of ranges, merge any overlapping ranges and normalize the
+    # them as a string that can be used directly on the switch.
+    #
+    # Note: The ranges are converted to ruby ranges for easy merging,
+    # then converted back to a cli-syntax ranges.
+    #
+    # Accepts an array or string:
+    #   ["2-5", "9", "4-6"]  -or-  '2-5, 9, 4-6'  -or-  ["2-5, 9, 4-6"]
+    # Returns a merged and ordered range:
+    #   ["2-6", "9"]
+    #
+    def self.normalize_range_array(range)
+      return range if range.nil? || range.empty?
+
+      # Handle string within an array: ["2-5, 9, 4-6"] to '2-5, 9, 4-6'
+      range = range.shift if range.is_a?(Array) && range.length == 1
+
+      # Handle string only: '2-5, 9, 4-6' to ["2-5", "9", "4-6"]
+      range = range.split(',') if range.is_a?(String)
```

Comment:
Technically speaking, I don't think you need the `if range.is_a?(String)` check.  Does not hurt anything, but range should always be a `String` at this point right?